src/nvm/buffer.rs

```diff
@@ -118,9 +124,11 @@ impl Buffer {
         let buffer_ref = buffer.get_ref();
 
         self.check_read_pos(start_pos)?;
```

Comment:
If we are reading a u16, we should check we are able to read start_pos + 1, so we are sure we can read until the end (2 bytes)

Reply:
We do not need to check for pos + 1, nom would do that and returns an error in case there is no melee data in the input buffer(pos + 1)